src/handle.ml

```diff
@@ -77,6 +77,49 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
   | P_tac_print
   | P_tac_proofterm
   | P_tac_focus(_)      -> assert false (* Handled above. *)
+  | P_tac_assert(t,h)     ->
+      (* Obtaining the goals environment, type and meta variable. *)
+      let env = fst (Proof.Goal.get_type g) in
+      let t = fst (Scope.scope_term StrMap.empty ss env t) in
+      let m = Proof.Goal.get_meta g in
+      (* Enriching the environnement with a var of type t *)
+      let h =
+        match h with
+        | None   -> "h"
+        | Some x -> x.elt
+      in
+      let var = Bindlib.new_var Terms.mkfree h in
+      let vs = Array.of_list (var::(List.map (fun (_,(x,_)) -> x) env)) in
+      (* The current goal is enriched with an hypothesis *)
+      let m2val =
+        match !(m.meta_value) with
+        | None     -> None
+        | Some(mb) ->
+           let mbox = Bindlib.box (snd (Bindlib.unmbind mb)) in
+           Some(Bindlib.unbox (Bindlib.bind_mvar vs mbox))
```

Comment:
Note that here, you are binding variables that have no reason to appear in `mbox` (if they did not already appear free in it before the call to `unbind` on the previous line). What were you trying to do here?